lib/ansible/module_utils/facts/collector.py

```diff
@@ -0,0 +1,134 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import sys
+
+from ansible.module_utils.facts.namespace import PrefixFactNamespace
+
+
+# TODO: BaseFactCollectors (plural) -> walks over list of collectors
+#       BaseFactCollector (singular) -> returns a dict (collectors 'leaf' node)
+#       and/or BaseFactCollectorNode etc
+# TODO/MAYBE: static/cls method of fact_id/tag matching? Someway for the gather spec
+#             matcher to handle semi dynamic names (like networks 'ansible_INTERFACENAME' facts)
+#             so gather could match them
+class BaseFactCollector:
+    _fact_ids = set([])
```

Comment:
Should be able to just do _fact_ids = set() without instantiating a temporary list.